Client/Frontend/Browser/Search/DefaultSearchPrefs.swift

```diff
@@ -59,13 +59,35 @@ class DefaultSearchPrefs {
         var usersEngineList = engineList ?? defaultSearchList
 
         // Append "all locales" search engines to the users engine list
-        usersEngineList = usersEngineList + allLocalesSearchList
+        usersEngineList += allLocalesSearchList
+        
+        // Append preferences
+        usersEngineList += filePaths(fromShortNames: preferences) ?? []
 
         // Overrides for specfic regions.
         if let overrides = regionOverrides[region].dictionary {
             usersEngineList = usersEngineList.map({ overrides[$0]?.string ?? $0 })
         }
-        return usersEngineList
+        
+        return usersEngineList.unique { $0 == $1 }
+    }
+    
+    private func filePaths(fromShortNames shortNames: [String]) -> [String]? {
+        guard let path = Bundle.main.path(forResource: "ShortNameToFileMapping", ofType: "json") else {
+            assertionFailure("Search list not found. Check bundle")
+            return nil
+        }
+        
+        do {
+            let filePath = URL(fileURLWithPath: path)
+            let jsonData = try Data(contentsOf: filePath)
+            let mappings = try JSONDecoder().decode([String: String].self, from: jsonData)
+            return shortNames.compactMap { mappings[$0] }
+        } catch {
+            // Fail silently, nothing needed for recovery here really
+        }
+        
+        return nil
```

Comment:
I would remove optional from method return, and return `[]` here so we don't need to unwrap it on line 65